redis/connection.py

```diff
@@ -502,6 +503,7 @@ def __init__(
         username=None,
         retry=None,
         redis_connect_func=None,
+        credential_provider: CredentialProvider = None,
```

Comment:
Considering the default is `None`, I believe this should be
```suggestion
        credential_provider: Optional[CredentialProvider] = None,
```